failed to run custom build command for `wasmedge-sys v0.5.0`

## Description

I am trying to embed the WasmEdge runtime in a Rust project that uses `actix-web`.

When running `cargo build` with `wasmedge-sys v0.5.0` as dependency, the build terminates on building `wasmedge-sys` and throws the following error:

```shell

failed to run custom build command for `wasmedge-sys v0.5.0`

Caused by:
  process didn't exit successfully: `C:\my\project\path\target\debug\build\wasmedge-sys-9e920e32a1f7ea85\build-script-build` (exit code: 101)
  --- stdout
  cargo:warning=[wasmedge-sys] not_standalone
  cargo:warning=[wasmedge-sys] TARGET_OS: windows
  cargo:warning=[wasmedge-sys] git init output: Output { status: ExitStatus(ExitStatus(0)), stdout: "Initialized empty Git repository in C:/my/project/path/target/debug/build/wasmedge-sys-56ae5bf345a3a2cf/out/wasmedge/.git/\n", stderr: "" }

cargo:warning=[wasmedge-sys] git remote add output: Output { status: ExitStatus(ExitStatus(0)), stdout: "", stderr: "" }
  cargo:warning=[wasmedge-sys] git fetch output: Output { status: ExitStatus(ExitStatus(0)), stdout: "", stderr: "From https://github.com/WasmEdge/WasmEdge\n * branch              HEAD       -> FETCH_HEAD\n" }
  cargo:warning=[wasmedge-sys] git checkout output: Output { status: ExitStatus(ExitStatus(0)), stdout: "", stderr: "Note: switching to 'FETCH_HEAD'.\n\nYou are in 'detached HEAD' state. You can look around, make experimental\nchanges and commit them, and you can discard any commits you make in this\nstate without impacting any branches by switching back to a branch.\n\nIf you want to create a new branch to retain commits you create, you may\ndo so (now or later) by using -c with the switch command. Example:\n\n  git switch -c <new-branch-name>\n\nOr undo this operation with:\n\n  git switch -\n\nTurn off this advice by setting config variable advice.detachedHead to false\n\nHEAD is now at 0c36dbe1 [Changelog] Update the changelog for the 0.10.0 release.\n" }

  --- stderr
  thread 'main' panicked at 'assertion failed: inc_dir.exists()', C:\Users\Me\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\build.rs:426:5  
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```


## Environment


- Hardware Architecture: x86/64
- Operating system: Windows 11

I have this in the dependencies in my `Cargo.toml` file:

```toml
[dependencies]
actix-web = "4"
wasmedge-sys = "0.5.0" # the same error occurs with 0.3.0 as well
wasmedge-types = "0.1.1"

```

## Steps to Reproduce

1. `cargo new <project-name>`
2. Add the above to your cargo dependencies
3. run `cargo build`

Can anyone help me with this? -- please

`clang` issue now resolved (thanks), but now I have this:

```shell

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ImportObjectGetModuleName` in module `ffi`
    --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\import_obj.rs:44:38
     |
44   |           let raw_name = unsafe { ffi::WasmEdge_ImportObjectGetModuleName(self.inner.0 as *const _) };
     |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `WasmEdge_ImportTypeGetModuleName`
     |
    ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:1391:5
     |
1391 | /     pub fn WasmEdge_ImportTypeGetModuleName(
1392 | |         Cxt: *const WasmEdge_ImportTypeContext,
1393 | |     ) -> WasmEdge_String;
     | |_________________________- similarly named function `WasmEdge_ImportTypeGetModuleName` defined here

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceGetModuleName` in module `ffi`
    --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:24:34
     |
24   |           let name = unsafe { ffi::WasmEdge_ModuleInstanceGetModuleName(self.inner.0 as *const _) };
     |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `WasmEdge_TableInstanceGetTableType`
     |
    ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:2580:5
     |
2580 | /     pub fn WasmEdge_TableInstanceGetTableType(
2581 | |         Cxt: *const WasmEdge_TableInstanceContext,
2582 | |     ) -> *const WasmEdge_TableTypeContext;
     | |__________________________________________- similarly named function `WasmEdge_TableInstanceGetTableType` defined here

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceFindFunction` in module `ffi`
  --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:46:18
   |
46 |             ffi::WasmEdge_ModuleInstanceFindFunction(
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ffi`

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceFindTable` in module `ffi`
  --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:78:18
   |
78 |             ffi::WasmEdge_ModuleInstanceFindTable(
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ffi`

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceFindMemory` in module `ffi`
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:108:18
    |
108 |             ffi::WasmEdge_ModuleInstanceFindMemory(
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ffi`

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceFindGlobal` in module `ffi`
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:138:18
    |
138 |             ffi::WasmEdge_ModuleInstanceFindGlobal(
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ffi`

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceListFunctionLength` in module `ffi`
    --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:157:23
     |
157  |         unsafe { ffi::WasmEdge_ModuleInstanceListFunctionLength(self.inner.0) }
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `WasmEdge_StoreListFunctionLength`
     |
    ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:2053:5
     |
2053 |     pub fn WasmEdge_StoreListFunctionLength(Cxt: *const WasmEdge_StoreContext) -> u32;
     |     ---------------------------------------------------------------------------------- similarly named function `WasmEdge_StoreListFunctionLength` defined here

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceListFunction` in module `ffi`
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:167:26
    |
167 |                     ffi::WasmEdge_ModuleInstanceListFunction(
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ffi`

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceListTableLength` in module `ffi`
    --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:187:23
     |
187  |         unsafe { ffi::WasmEdge_ModuleInstanceListTableLength(self.inner.0) }
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `WasmEdge_StoreListTableLength`
     |
    ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:2126:5
     |
2126 |     pub fn WasmEdge_StoreListTableLength(Cxt: *const WasmEdge_StoreContext) -> u32;
     |     ------------------------------------------------------------------------------- similarly named function `WasmEdge_StoreListTableLength` defined here

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceListTable` in module `ffi`
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:197:26
    |
197 |                     ffi::WasmEdge_ModuleInstanceListTable(
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ffi`

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceListMemoryLength` in module `ffi`
    --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:217:23
     |
217  |         unsafe { ffi::WasmEdge_ModuleInstanceListMemoryLength(self.inner.0) }
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `WasmEdge_StoreListMemoryLength`
     |
    ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:2199:5
     |
2199 |     pub fn WasmEdge_StoreListMemoryLength(Cxt: *const WasmEdge_StoreContext) -> u32;
     |     -------------------------------------------------------------------------------- similarly named function `WasmEdge_StoreListMemoryLength` defined here

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceListMemory` in module `ffi`
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:227:26
    |
227 |                     ffi::WasmEdge_ModuleInstanceListMemory(
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ffi`

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceListGlobalLength` in module `ffi`
    --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:247:23
     |
247  |         unsafe { ffi::WasmEdge_ModuleInstanceListGlobalLength(self.inner.0) }
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `WasmEdge_StoreListGlobalLength`
     |
    ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:2272:5
     |
2272 |     pub fn WasmEdge_StoreListGlobalLength(Cxt: *const WasmEdge_StoreContext) -> u32;
     |     -------------------------------------------------------------------------------- similarly named function `WasmEdge_StoreListGlobalLength` defined here

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ModuleInstanceListGlobal` in module `ffi`
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:257:26
    |
257 |                     ffi::WasmEdge_ModuleInstanceListGlobal(
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ffi`

error[E0412]: cannot find type `WasmEdge_ModuleInstanceContext` in module `ffi`
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\instance\module.rs:277:56
    |
277 | pub(crate) struct InnerInstance(pub(crate) *const ffi::WasmEdge_ModuleInstanceContext);
    |                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a struct with a similar name exists: `WasmEdge_TableInstanceContext`
    |
   ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:445:1
    |
445 | pub struct WasmEdge_TableInstanceContext {
    | ---------------------------------------- similarly named struct `WasmEdge_TableInstanceContext` defined here

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_StoreGetActiveModule` in module `ffi`
    --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\store.rs:618:33
     |
618  |           let ctx = unsafe { ffi::WasmEdge_StoreGetActiveModule(self.inner.0) };
     |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `WasmEdge_StoreListModule`
     |
    ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:2358:5
     |
2358 | /     pub fn WasmEdge_StoreListModule(
2359 | |         Cxt: *const WasmEdge_StoreContext,
2360 | |         Names: *mut WasmEdge_String,
2361 | |         Len: u32,
2362 | |     ) -> u32;
     | |_____________- similarly named function `WasmEdge_StoreListModule` defined here

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_StoreFindModule` in module `ffi`
    --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\store.rs:639:33
     |
639  |           let ctx = unsafe { ffi::WasmEdge_StoreFindModule(self.inner.0, mod_name.as_raw()) };
     |                                   ^^^^^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `WasmEdge_StoreListModule`
     |
    ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:2358:5
     |
2358 | /     pub fn WasmEdge_StoreListModule(
2359 | |         Cxt: *const WasmEdge_StoreContext,
2360 | |         Names: *mut WasmEdge_String,
2361 | |         Len: u32,
2362 | |     ) -> u32;
     | |_____________- similarly named function `WasmEdge_StoreListModule` defined here

error[E0425]: cannot find value `WasmEdge_ValType_None` in module `ffi`
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:107:35
    |
107 |             ValType::None => ffi::WasmEdge_ValType_None,
    |                                   ^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `WasmEdge_ValType_F32`
    |
   ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:174:1
    |
174 | pub const WasmEdge_ValType_F32: WasmEdge_ValType = 125;
    | ------------------------------------------------------- similarly named constant `WasmEdge_ValType_F32` defined here

error[E0531]: cannot find unit struct, unit variant or constant `WasmEdge_ValType_None` in module `ffi`
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:121:18
    |
121 |             ffi::WasmEdge_ValType_None => ValType::None,
    |                  ^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `WasmEdge_ValType_F32`
    |
   ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:174:1
    |
174 | pub const WasmEdge_ValType_F32: WasmEdge_ValType = 125;
    | ------------------------------------------------------- similarly named constant `WasmEdge_ValType_F32` defined here

error[E0425]: cannot find function, tuple struct or tuple variant `WasmEdge_ValueGetFuncRef` in module `ffi`
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:477:36
    |
477 |                     let ctx = ffi::WasmEdge_ValueGetFuncRef(self.ctx);
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `WasmEdge_ValueGenFuncRef`
    |
   ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:569:5
    |
569 |     pub fn WasmEdge_ValueGenFuncRef(Index: u32) -> WasmEdge_Value;
    |     -------------------------------------------------------------- similarly named function `WasmEdge_ValueGenFuncRef` defined here

error[E0531]: cannot find unit struct, unit variant or constant `WasmEdge_ValType_None` in module `ffi`
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:552:18
    |
552 |             ffi::WasmEdge_ValType_None => Self {
    |                  ^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `WasmEdge_ValType_F32`
    |
   ::: C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-56ae5bf345a3a2cf\out/wasmedge.rs:174:1
    |
174 | pub const WasmEdge_ValType_F32: WasmEdge_ValType = 125;
    | ------------------------------------------------------- similarly named constant `WasmEdge_ValType_F32` defined here

error[E0308]: mismatched types
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:160:10
    |
160 |     O0 = ffi::WasmEdge_CompilerOptimizationLevel_O0,
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`

error[E0308]: mismatched types
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:163:10
    |
163 |     O1 = ffi::WasmEdge_CompilerOptimizationLevel_O1,
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`

error[E0308]: mismatched types
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:167:10
    |
167 |     O2 = ffi::WasmEdge_CompilerOptimizationLevel_O2,
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`

error[E0308]: mismatched types
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:170:10
    |
170 |     O3 = ffi::WasmEdge_CompilerOptimizationLevel_O3,
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`

error[E0308]: mismatched types
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:174:10
    |
174 |     Os = ffi::WasmEdge_CompilerOptimizationLevel_Os,
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`

error[E0308]: mismatched types
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:177:10
    |
177 |     Oz = ffi::WasmEdge_CompilerOptimizationLevel_Oz,
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`

error[E0308]: mismatched types
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:198:14
    |
198 |     Native = ffi::WasmEdge_CompilerOutputFormat_Native,
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`

error[E0308]: mismatched types
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:201:12
    |
201 |     Wasm = ffi::WasmEdge_CompilerOutputFormat_Wasm,
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`

error[E0308]: mismatched types
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:243:16
    |
243 |     Function = ffi::WasmEdge_ExternalType_Function,
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`

error[E0308]: mismatched types
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:244:13
    |
244 |     Table = ffi::WasmEdge_ExternalType_Table,
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`

error[E0308]: mismatched types
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:245:14
    |
245 |     Memory = ffi::WasmEdge_ExternalType_Memory,
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`

error[E0308]: mismatched types
   --> C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.5.0\src\types.rs:246:14
    |
246 |     Global = ffi::WasmEdge_ExternalType_Global,
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`

```
Even after pulling fresh `v0.5.0` code into my `cargo` registry. I should also mention that I am nothing close to a `rust` expert. 
Thanks.

So I upgraged `wasmedge-sys` to `v0.7.0`  and `wasmedge` to `v0.10.0` as @apepkuss suggested. Things seemed to go a bit better than now except that I encountered this:

```shell
error: failed to run custom build command for `wasmedge-sys v0.7.0`

Caused by:
  process didn't exit successfully: `C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-1c84e13f0b485641\build-script-build` (exit code: 101)
  --- stdout
  cargo:warning=[wasmedge-sys] Use WASMEDGE_INCLUDE_DIR: "C:\\Program Files\\WasmEdge\\include"
  cargo:warning=[wasmedge-sys] Use WASMEDGE_LIB_DIR: "C:\\Program Files\\WasmEdge\\lib"

  --- stderr
  C:\Program Files\WasmEdge\include\wasmedge/enum_configure.h:59:10: fatal error: 'enum.inc' file not found
  C:\Program Files\WasmEdge\include\wasmedge/enum_configure.h:59:10: fatal error: 'enum.inc' file not found, err: true
  thread 'main' panicked at 'failed to generate bindings: ()', C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.7.0\build.rs:51:10
```
 Any suggestions, please?

> So I upgraged `wasmedge-sys` to `v0.7.0` and `wasmedge` to `v0.10.0` as @apepkuss suggested. Things seemed to have gone a bit better, except that I encountered this:
> 
> ```shell
> error: failed to run custom build command for `wasmedge-sys v0.7.0`
> 
> Caused by:
>   process didn't exit successfully: `C:\Users\Farouk\Desktop\exprj\more recent\werust\target\debug\build\wasmedge-sys-1c84e13f0b485641\build-script-build` (exit code: 101)
>   --- stdout
>   cargo:warning=[wasmedge-sys] Use WASMEDGE_INCLUDE_DIR: "C:\\Program Files\\WasmEdge\\include"
>   cargo:warning=[wasmedge-sys] Use WASMEDGE_LIB_DIR: "C:\\Program Files\\WasmEdge\\lib"
> 
>   --- stderr
>   C:\Program Files\WasmEdge\include\wasmedge/enum_configure.h:59:10: fatal error: 'enum.inc' file not found
>   C:\Program Files\WasmEdge\include\wasmedge/enum_configure.h:59:10: fatal error: 'enum.inc' file not found, err: true
>   thread 'main' panicked at 'failed to generate bindings: ()', C:\Users\Farouk\.cargo\registry\src\github.com-1ecc6299db9ec823\wasmedge-sys-0.7.0\build.rs:51:10
> ```
> 
> Any suggestions, please?

@hydai Could you help with this issue? It seems a failure was caused during the compilation of `WasmEdge`. It behaves so weirdly on the Windows platform.

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -8,6 +8,7 @@ on:
   push:
     branches:
       - master
+      - fix-ci-windows
     paths:
       - ".github/workflows/bindings-rust.yml"
       - "bindings/rust/**"
@@ -175,7 +176,7 @@ jobs:
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           $llvm = "LLVM-13.0.1-win64.zip"
           curl -sLO https://github.com/WasmEdge/llvm-windows/releases/download/llvmorg-13.0.1/LLVM-13.0.1-win64.zip -o $llvm
-          Expand-Archive -Path $llvm
+          Expand-Archive -Path $llvm -Force
           $llvm_dir = "$pwd\\LLVM-13.0.1-win64\\LLVM-13.0.1-win64\\lib\\cmake\\llvm"
           $Env:CC = "clang-cl"
           $Env:CXX = "clang-cl"

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -219,7 +219,8 @@ jobs:
           Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\include\api\wasmedge -Force
           Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\lib\api -Force
           Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\plugins -Force
-          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\tools -Force
+          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\tools\wasmedge -Force
+          Get-ChildItem -Path C:\Users\runneradmin\.rustup\toolchains -Force
           $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin;D:\a\WasmEdge-1\WasmEdge-1\build\lib\api"
           (type env:path) -split ';'
           # dir env:

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -8,7 +8,6 @@ on:
   push:
     branches:
       - master
-      - fix-ci-windows
     paths:
       - ".github/workflows/bindings-rust.yml"
       - "bindings/rust/**"

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -213,13 +213,6 @@ jobs:
           $vsPath = (vswhere -latest -property installationPath)
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
-          rustc --print sysroot
-          Get-Location
-          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\include\api\wasmedge -Force
-          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\lib\api -Force
-          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\plugins -Force
-          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\tools\wasmedge -Force
-          Get-ChildItem -Path C:\Users\runneradmin\.rustup\toolchains -Force
-          # $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;D:\a\WasmEdge-1\WasmEdge-1\build\lib\api"
-          # cd bindings\rust
-          # cargo +nightly test --lib --examples --locked --verbose
+          $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;D:\a\WasmEdge\WasmEdge\build\lib\api"
+          cd bindings\rust
+          cargo +nightly test --lib --examples --locked --verbose

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -216,7 +216,7 @@ jobs:
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           rustc --print sysroot 
           # echo "C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
-          # (type env:path) -split ';'
           $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc"
+          (type env:path) -split ';'
           cd bindings/rust/
           cargo +nightly test --lib --examples --locked

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -216,7 +216,7 @@ jobs:
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           rustc --print sysroot 
           # echo "C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
-          $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin"
+          $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin"
           (type env:path) -split ';'
           cd bindings/rust/
           cargo +nightly test --lib --examples --locked

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -215,6 +215,7 @@ jobs:
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           rustc --print sysroot 
-          echo "C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
-          cd bindings/rust/
-          cargo +nightly test --lib --examples --locked
+          # echo "C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+          (type env:path) -split ';'
+          # cd bindings/rust/
+          # cargo +nightly test --lib --examples --locked

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -221,8 +221,8 @@ jobs:
           Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\plugins -Force
           Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\tools\wasmedge -Force
           Get-ChildItem -Path C:\Users\runneradmin\.rustup\toolchains -Force
-          $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin;D:\a\WasmEdge-1\WasmEdge-1\build\lib\api"
+          $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;D:\a\WasmEdge-1\WasmEdge-1\build\lib\api"
           (type env:path) -split ';'
           # dir env:
-          # cd bindings\rust
-          # cargo +nightly test --lib --examples --locked --verbose
+          cd bindings\rust
+          cargo +nightly test --lib --examples --locked --verbose

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -147,7 +147,7 @@ jobs:
     env:
       WASMEDGE_DIR: ${{ github.workspace }}
       WASMEDGE_BUILD_DIR: ${{ github.workspace }}\build
-      WASMEDGE_PLUGIN_PATH: ${{ github.workspace }}\build\wasmedge\plugins\wasmedge_process
+      # WASMEDGE_PLUGIN_PATH: ${{ github.workspace }}\build\wasmedge\plugins\wasmedge_process
       LD_LIBRARY_PATH: ${{ github.workspace }}\build\lib\api
     steps:
       - uses: actions/checkout@v2
@@ -216,9 +216,10 @@ jobs:
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           # Get-Location
           Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\bin -Force
-          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\include\api -Force
-          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\lib -Force
+          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\include\api\wasmedge -Force
+          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\lib\api -Force
           Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\plugins -Force
+          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\tools -Force
           # $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin"
           # (type env:path) -split ';'
           # dir env:

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -157,7 +157,7 @@ jobs:
       - name: Install dependency
         uses: crazy-max/ghaction-chocolatey@v1
         with:
-          args: install cmake ninja vswhere
+          args: install cmake ninja vswhere --allow-unofficial
 
       - uses: GuillaumeFalourd/setup-windows10-sdk-action@v1
         with:

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -215,13 +215,13 @@ jobs:
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           # Get-Location
-          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\bin -Force
+          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\bin -Force
           Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\include\api\wasmedge -Force
           Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\lib\api -Force
           Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\plugins -Force
           Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\tools -Force
-          # $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin"
-          # (type env:path) -split ';'
+          $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin;D:\a\WasmEdge-1\WasmEdge-1\build\lib\api"
+          (type env:path) -split ';'
           # dir env:
           # cd bindings\rust
           # cargo +nightly test --lib --examples --locked --verbose

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -216,4 +216,3 @@ jobs:
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           cd bindings/rust/
           cargo +nightly test --lib --examples --locked
-          cargo +nightly test --doc --locked

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -214,14 +214,6 @@ jobs:
           $vsPath = (vswhere -latest -property installationPath)
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
-          # Get-Location
-          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\bin -Force
-          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\include\api\wasmedge -Force
-          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\lib\api -Force
-          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\plugins -Force
-          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\tools\wasmedge -Force
-          # Get-ChildItem -Path C:\Users\runneradmin\.rustup\toolchains -Force
           $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;D:\a\WasmEdge-1\WasmEdge-1\build\lib\api"
-          # (type env:path) -split ';'
           cd bindings\rust
           cargo +nightly test --lib --examples --locked --verbose

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -214,11 +214,11 @@ jobs:
           $vsPath = (vswhere -latest -property installationPath)
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
-          Get-Location
-          dir build\bin
-          dir build\include
-          dir build\lib
-          dir build\plugins
+          # Get-Location
+          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\bin -Force
+          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\include\api -Force
+          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\lib -Force
+          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\plugins -Force
           # $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin"
           # (type env:path) -split ';'
           # dir env:

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -214,9 +214,9 @@ jobs:
           $vsPath = (vswhere -latest -property installationPath)
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
-          rustc --print sysroot 
-          # echo "C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
-          $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin"
-          (type env:path) -split ';'
-          cd bindings/rust/
-          cargo +nightly test --lib --examples --locked
+          Get-Location
+          # $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin"
+          # (type env:path) -split ';'
+          dir env:
+          # cd bindings/rust/
+          # cargo +nightly test --lib --examples --locked

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -216,7 +216,7 @@ jobs:
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           rustc --print sysroot 
           # echo "C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
-          $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc"
+          $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin"
           (type env:path) -split ';'
-          # cd bindings/rust/
-          # cargo +nightly test --lib --examples --locked
+          cd bindings/rust/
+          cargo +nightly test --lib --examples --locked

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -215,9 +215,12 @@ jobs:
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           Get-Location
-          dir build
+          dir build\bin
+          dir build\include
+          dir build\lib
+          dir build\plugins
           # $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin"
           # (type env:path) -split ';'
           # dir env:
-          # cd bindings/rust/
-          # cargo +nightly test --lib --examples --locked
+          # cd bindings\rust
+          # cargo +nightly test --lib --examples --locked --verbose

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -169,33 +169,6 @@ jobs:
         with:
           sdk-version: 19041
 
-      - name: Build WasmEdge
-        run: |
-          $vsPath = (vswhere -latest -property installationPath)
-          Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
-          Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
-          $llvm = "LLVM-13.0.1-win64.zip"
-          curl -sLO https://github.com/WasmEdge/llvm-windows/releases/download/llvmorg-13.0.1/LLVM-13.0.1-win64.zip -o $llvm
-          Expand-Archive -Path $llvm -Force
-          $llvm_dir = "$pwd\\LLVM-13.0.1-win64\\LLVM-13.0.1-win64\\lib\\cmake\\llvm"
-          $Env:CC = "clang-cl"
-          $Env:CXX = "clang-cl"
-          $cmake_sys_version = "10.0.19041.0"
-          cmake -Bbuild -GNinja -DCMAKE_BUILD_TYPE=Release "-DCMAKE_SYSTEM_VERSION=$cmake_sys_version" -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreadedDLL "-DLLVM_DIR=$llvm_dir" -DWASMEDGE_BUILD_TESTS=ON -DWASMEDGE_BUILD_PACKAGE="ZIP" .
-          cmake --build build
-
-      - name: Test WasmEdge
-        continue-on-error: true
-        run: |
-          $vsPath = (vswhere -latest -property installationPath)
-          Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
-          Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
-          $Env:PATH += ";$pwd\\build\\lib\\api"
-          cd build
-          tools\wasmedge\wasmedge -v
-          ctest
-          cd -
-
       - name: Install Nightly Rust
         uses: actions-rs/toolchain@v1
         with:

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -215,6 +215,6 @@ jobs:
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           rustc --print sysroot 
-          set PATH=%PATH%;C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin
+          set PATH="%PATH%;C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin"
           cd bindings/rust/
           cargo +nightly test --lib --examples --locked

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -216,13 +216,12 @@ jobs:
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           # Get-Location
           # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\bin -Force
-          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\include\api\wasmedge -Force
-          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\lib\api -Force
-          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\plugins -Force
-          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\tools\wasmedge -Force
-          Get-ChildItem -Path C:\Users\runneradmin\.rustup\toolchains -Force
+          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\include\api\wasmedge -Force
+          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\lib\api -Force
+          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\plugins -Force
+          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\tools\wasmedge -Force
+          # Get-ChildItem -Path C:\Users\runneradmin\.rustup\toolchains -Force
           $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;D:\a\WasmEdge-1\WasmEdge-1\build\lib\api"
-          (type env:path) -split ';'
-          # dir env:
+          # (type env:path) -split ';'
           cd bindings\rust
           cargo +nightly test --lib --examples --locked --verbose

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -215,4 +215,5 @@ jobs:
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           cd bindings/rust/
-          cargo +nightly test --lib --examples --locked
+          rustc --print sysroot 
+          # cargo +nightly test --lib --examples --locked

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -215,10 +215,10 @@ jobs:
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           rustc --print sysroot
           Get-Location
-          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\include\api\wasmedge -Force
-          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\lib\api -Force
-          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\plugins -Force
-          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\tools\wasmedge -Force
+          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\include\api\wasmedge -Force
+          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\lib\api -Force
+          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\plugins -Force
+          # Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\tools\wasmedge -Force
           Get-ChildItem -Path C:\Users\runneradmin\.rustup\toolchains -Force
           # $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;D:\a\WasmEdge-1\WasmEdge-1\build\lib\api"
           # cd bindings\rust

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -216,6 +216,7 @@ jobs:
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           rustc --print sysroot 
           # echo "C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
-          (type env:path) -split ';'
-          # cd bindings/rust/
-          # cargo +nightly test --lib --examples --locked
+          # (type env:path) -split ';'
+          $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc"
+          cd bindings/rust/
+          cargo +nightly test --lib --examples --locked

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -74,7 +74,7 @@ jobs:
           cargo +nightly clippy --profile test --lib --examples -- -D warnings -D clippy::dbg_macro
           cd -
 
-      - name: Test
+      - name: Test Rust Bindings
         run: |
           cd bindings/rust/
           export WASMEDGE_DIR="$(pwd)/../../"
@@ -98,12 +98,6 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Install stable toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          components: rustfmt, clippy
-
       - name: Install nightly toolchain
         uses: actions-rs/toolchain@v1
         with:
@@ -116,7 +110,7 @@ jobs:
           cargo +nightly fmt --all -- --check
           cd -
 
-      - name: Build WasmEdge
+      - name: Build WasmEdge with Release mode
         run: |
           brew install llvm ninja boost cmake
           export PATH="/usr/local/opt/llvm/bin:$PATH"
@@ -124,7 +118,7 @@ jobs:
           export CPPFLAGS="-I/usr/local/opt/llvm/include"
           export CC=clang
           export CXX=clang++
-          cmake -Bbuild -GNinja -DWASMEDGE_BUILD_TESTS=ON -DWASMEDGE_BUILD_PACKAGE="TGZ" .
+          cmake -Bbuild -GNinja -DCMAKE_BUILD_TYPE=Release -DWASMEDGE_BUILD_TESTS=ON -DWASMEDGE_BUILD_PACKAGE="TGZ" .
           cmake --build build
 
       - name: Clippy
@@ -136,7 +130,7 @@ jobs:
           cargo +nightly clippy --profile test --lib --examples -- -D warnings -D clippy::dbg_macro
           cd -
 
-      - name: Test
+      - name: Test Rust Bindings
         run: |
           cd bindings/rust/
           export WASMEDGE_DIR="$(pwd)/../../"
@@ -180,7 +174,7 @@ jobs:
           cd bindings/rust/
           cargo +nightly fmt --all -- --check
 
-      - name: Build WasmEdge
+      - name: Build WasmEdge with Release mode
         run: |
           $vsPath = (vswhere -latest -property installationPath)
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
@@ -215,7 +209,7 @@ jobs:
           cd bindings/rust/
           cargo +nightly clippy --verbose --profile test --lib --examples -- -D warnings -D clippy::dbg_macro
 
-      - name: Test
+      - name: Test Rust Bindings
         run: |
           $vsPath = (vswhere -latest -property installationPath)
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -213,6 +213,13 @@ jobs:
           $vsPath = (vswhere -latest -property installationPath)
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
-          $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;D:\a\WasmEdge-1\WasmEdge-1\build\lib\api"
-          cd bindings\rust
-          cargo +nightly test --lib --examples --locked --verbose
+          rustc --print sysroot
+          Get-Location
+          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\include\api\wasmedge -Force
+          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\lib\api -Force
+          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\plugins -Force
+          Get-ChildItem -Path D:\a\WasmEdge-1\WasmEdge-1\build\tools\wasmedge -Force
+          Get-ChildItem -Path C:\Users\runneradmin\.rustup\toolchains -Force
+          # $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\bin;D:\a\WasmEdge-1\WasmEdge-1\build\lib\api"
+          # cd bindings\rust
+          # cargo +nightly test --lib --examples --locked --verbose

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -147,7 +147,7 @@ jobs:
     env:
       WASMEDGE_DIR: ${{ github.workspace }}
       WASMEDGE_BUILD_DIR: ${{ github.workspace }}\build
-      # WASMEDGE_PLUGIN_PATH: ${{ github.workspace }}\build\wasmedge\plugins\wasmedge_process
+      WASMEDGE_PLUGIN_PATH: ${{ github.workspace }}\build\wasmedge\plugins\wasmedge_process
       LD_LIBRARY_PATH: ${{ github.workspace }}\build\lib\api
     steps:
       - uses: actions/checkout@v2
@@ -157,7 +157,7 @@ jobs:
       - name: Install dependency
         uses: crazy-max/ghaction-chocolatey@v1
         with:
-          args: install cmake ninja vswhere --allow-unofficial
+          args: install cmake ninja vswhere
 
       - uses: GuillaumeFalourd/setup-windows10-sdk-action@v1
         with:

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -215,6 +215,6 @@ jobs:
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           rustc --print sysroot 
-          set PATH="%PATH%;C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin"
+          echo "C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
           cd bindings/rust/
           cargo +nightly test --lib --examples --locked

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -215,8 +215,9 @@ jobs:
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
           Get-Location
+          dir build
           # $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin"
           # (type env:path) -split ';'
-          dir env:
+          # dir env:
           # cd bindings/rust/
           # cargo +nightly test --lib --examples --locked

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -218,5 +218,5 @@ jobs:
           # echo "C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
           $env:Path="$env:Path;C:\Users\runneradmin\.rustup\toolchains\stable-x86_64-pc-windows-msvc"
           (type env:path) -split ';'
-          cd bindings/rust/
-          cargo +nightly test --lib --examples --locked
+          # cd bindings/rust/
+          # cargo +nightly test --lib --examples --locked

--- a/.github/workflows/bindings-rust.yml
+++ b/.github/workflows/bindings-rust.yml
@@ -214,6 +214,7 @@ jobs:
           $vsPath = (vswhere -latest -property installationPath)
           Import-Module (Join-Path $vsPath "Common7\Tools\Microsoft.VisualStudio.DevShell.dll")
           Enter-VsDevShell -VsInstallPath $vsPath -SkipAutomaticLocation -DevCmdArguments "-arch=x64 -host_arch=x64 -winsdk=10.0.19041.0"
-          cd bindings/rust/
           rustc --print sysroot 
-          # cargo +nightly test --lib --examples --locked
+          set PATH=%PATH%;C:\Users\runneradmin\.rustup\toolchains\%RUST_VERSION%-%TARGET%\bin
+          cd bindings/rust/
+          cargo +nightly test --lib --examples --locked

--- a/bindings/rust/wasmedge-sdk/src/import.rs
+++ b/bindings/rust/wasmedge-sdk/src/import.rs
@@ -376,6 +376,7 @@ mod tests {
     }
 
     #[test]
+    #[cfg(unix)]
     fn test_import_builder_wasi() {
         let result = ImportObjectBuilder::default().build_as_wasi(None, None, None);
         assert!(result.is_ok());

--- a/bindings/rust/wasmedge-sdk/src/lib.rs
+++ b/bindings/rust/wasmedge-sdk/src/lib.rs
@@ -10,10 +10,16 @@
 //!
 //! The [wasmedge-sdk](https://crates.io/crates/wasmedge-sdk) crate defines a group of high-level Rust APIs, which are used to build up business applications.
 //!
-//! Notice that
-//! - `WasmEdge Rust SDK` uses nightly version of Rust.
+//! Notice that `WasmEdge Rust SDK` uses nightly version of Rust. It's strongly recommended to use the latest nightly version of Rust.
 //!
-//! - Due to [issue #1527](https://github.com/WasmEdge/WasmEdge/issues/1527), `WasmEdge Rust SDK` cannot build successfully on Windows platform. Please [use Docker](https://wasmedge.org/book/en/start/docker.html) to build `WasmEdge Rust SDK` on Windows.
+//! ## Versioning Table
+//!
+//! The following table provides the versioning information about each crate of WasmEdge Rust bindings.
+//!
+//! | wasmedge-sdk  | WasmEdge lib                                                       | wasmedge-sys  | wasmedge-types|
+//! | :-----------: | :----------------------------------------------------------------: | :-----------: | :-----------: |
+//! | 0.2.0         | [0.10.1](https://github.com/WasmEdge/WasmEdge/releases/tag/0.10.0) | 0.8           | 0.2           |
+//! | 0.1.0         | [0.10.0](https://github.com/WasmEdge/WasmEdge/releases/tag/0.10.0) | 0.7           | 0.1           |
 //!
 //! ## Usage
 //!

--- a/bindings/rust/wasmedge-sys/src/config.rs
+++ b/bindings/rust/wasmedge-sys/src/config.rs
@@ -645,7 +645,10 @@ impl Config {
     #[cfg(feature = "aot")]
     pub fn set_aot_optimization_level(&mut self, opt_level: CompilerOptimizationLevel) {
         unsafe {
-            ffi::WasmEdge_ConfigureCompilerSetOptimizationLevel(self.inner.0, opt_level as u32)
+            ffi::WasmEdge_ConfigureCompilerSetOptimizationLevel(
+                self.inner.0,
+                opt_level as ffi::WasmEdge_CompilerOptimizationLevel,
+            )
         }
     }
 
@@ -667,7 +670,12 @@ impl Config {
     /// * `format` - The format of the output binary.
     #[cfg(feature = "aot")]
     pub fn set_aot_compiler_output_format(&mut self, format: CompilerOutputFormat) {
-        unsafe { ffi::WasmEdge_ConfigureCompilerSetOutputFormat(self.inner.0, format as u32) }
+        unsafe {
+            ffi::WasmEdge_ConfigureCompilerSetOutputFormat(
+                self.inner.0,
+                format as ffi::WasmEdge_CompilerOutputFormat,
+            )
+        }
     }
 
     /// Returns the output binary format of AOT compiler.

--- a/bindings/rust/wasmedge-sys/src/instance/module.rs
+++ b/bindings/rust/wasmedge-sys/src/instance/module.rs
@@ -1492,6 +1492,7 @@ mod tests {
     }
 
     #[test]
+    #[cfg(unix)]
     fn test_instance_wasi() {
         // create a wasi module instance
         {

--- a/bindings/rust/wasmedge-sys/src/lib.rs
+++ b/bindings/rust/wasmedge-sys/src/lib.rs
@@ -8,10 +8,16 @@
 //!
 //! For developers, it is strongly recommended that the APIs in `wasmedge-sys` are used to construct high-level libraries, while `wasmedge-sdk` is for building up business applications.
 //!
-//! Notice that
-//! - `WasmEdge Rust SDK` uses nightly version of Rust.
+//! Notice that `WasmEdge Rust SDK` uses nightly version of Rust. It's strongly recommended to use the latest nightly version of Rust.
 //!
-//! - Due to [issue #1527](https://github.com/WasmEdge/WasmEdge/issues/1527), `WasmEdge Rust SDK` cannot build successfully on Windows platform. Please [use Docker](https://wasmedge.org/book/en/start/docker.html) to build `WasmEdge Rust SDK` on Windows.
+//! ## Versioning Table
+//!
+//! The following table provides the versioning information about each crate of WasmEdge Rust bindings.
+//!
+//! | wasmedge-sdk  | WasmEdge lib                                                       | wasmedge-sys  | wasmedge-types|
+//! | :-----------: | :----------------------------------------------------------------: | :-----------: | :-----------: |
+//! | 0.2.0         | [0.10.1](https://github.com/WasmEdge/WasmEdge/releases/tag/0.10.0) | 0.8           | 0.2           |
+//! | 0.1.0         | [0.10.0](https://github.com/WasmEdge/WasmEdge/releases/tag/0.10.0) | 0.7           | 0.1           |
 //!
 //! ## Usage
 //!

--- a/bindings/rust/wasmedge-sys/src/types.rs
+++ b/bindings/rust/wasmedge-sys/src/types.rs
@@ -343,11 +343,13 @@ impl From<ffi::WasmEdge_Value> for WasmValue {
 #[cfg(test)]
 mod tests {
     use super::*;
+    #[cfg(unix)]
     use crate::{Table, TableType};
     use std::{
         sync::{Arc, Mutex},
         thread,
     };
+    #[cfg(unix)]
     use wasmedge_types::RefType;
 
     #[test]

--- a/bindings/rust/wasmedge-sys/src/types.rs
+++ b/bindings/rust/wasmedge-sys/src/types.rs
@@ -351,6 +351,7 @@ mod tests {
     use wasmedge_types::RefType;
 
     #[test]
+    #[cfg(unix)]
     fn test_types_value() {
         // I32
         let val = WasmValue::from_i32(1314);
@@ -401,6 +402,7 @@ mod tests {
     }
 
     #[test]
+    #[cfg(unix)]
     fn test_types_value_send() {
         // I32
         let val_i32 = WasmValue::from_i32(1314);
@@ -463,6 +465,7 @@ mod tests {
     }
 
     #[test]
+    #[cfg(unix)]
     fn test_types_value_sync() {
         // I32
         let val_i32 = Arc::new(Mutex::new(WasmValue::from_i32(1314)));

--- a/bindings/rust/wasmedge-sys/src/utils.rs
+++ b/bindings/rust/wasmedge-sys/src/utils.rs
@@ -13,7 +13,7 @@ use std::{
     path::Path,
 };
 
-// #[cfg(unix)]
+#[cfg(unix)]
 pub(crate) fn path_to_cstring(path: &Path) -> WasmEdgeResult<CString> {
     use std::os::unix::ffi::OsStrExt;
     Ok(CString::new(path.as_os_str().as_bytes())?)

--- a/bindings/rust/wasmedge-sys/src/utils.rs
+++ b/bindings/rust/wasmedge-sys/src/utils.rs
@@ -13,21 +13,9 @@ use std::{
     path::Path,
 };
 
-#[cfg(unix)]
 pub(crate) fn path_to_cstring(path: &Path) -> WasmEdgeResult<CString> {
     use std::os::unix::ffi::OsStrExt;
     Ok(CString::new(path.as_os_str().as_bytes())?)
-}
-
-#[cfg(windows)]
-pub(crate) fn path_to_cstring(path: &Path) -> WasmEdgeResult<CString> {
-    match path.to_str() {
-        Some(s) => Ok(CString::new(s)?),
-        None => {
-            let message = format!("Couldn't convert path '{}' to UTF-8", path.display());
-            Err(message.into())
-        }
-    }
 }
 
 /// Logs the debug information.

--- a/bindings/rust/wasmedge-sys/src/utils.rs
+++ b/bindings/rust/wasmedge-sys/src/utils.rs
@@ -13,9 +13,20 @@ use std::{
     path::Path,
 };
 
+// #[cfg(unix)]
 pub(crate) fn path_to_cstring(path: &Path) -> WasmEdgeResult<CString> {
     use std::os::unix::ffi::OsStrExt;
     Ok(CString::new(path.as_os_str().as_bytes())?)
+}
+
+#[cfg(windows)]
+pub(crate) fn path_to_cstring(path: &Path) -> WasmEdgeResult<CString> {
+    match path.to_str() {
+        Some(s) => Ok(CString::new(s)?),
+        None => Err(WasmEdgeError::WindowsPathConversion(
+            path.to_string_lossy().to_string(),
+        )),
+    }
 }
 
 /// Logs the debug information.

--- a/bindings/rust/wasmedge-sys/src/vm.rs
+++ b/bindings/rust/wasmedge-sys/src/vm.rs
@@ -2046,6 +2046,7 @@ mod tests {
     }
 
     #[test]
+    #[cfg(unix)]
     fn test_vm_get_wasi_module() {
         {
             // create a Config context

--- a/bindings/rust/wasmedge-sys/src/vm.rs
+++ b/bindings/rust/wasmedge-sys/src/vm.rs
@@ -2301,6 +2301,7 @@ mod tests {
         result.unwrap()
     }
 
+    #[cfg(unix)]
     fn real_add(inputs: Vec<WasmValue>) -> Result<Vec<WasmValue>, u8> {
         if inputs.len() != 2 {
             return Err(1);

--- a/bindings/rust/wasmedge-sys/src/vm.rs
+++ b/bindings/rust/wasmedge-sys/src/vm.rs
@@ -890,13 +890,16 @@ unsafe impl Sync for InnerVm {}
 #[cfg(test)]
 mod tests {
     use super::Vm;
+    #[cfg(unix)]
+    use crate::{
+        error::CoreInstantiationError, FuncType, Function, ImportInstance, ImportObject, WasiModule,
+    };
     use crate::{
         error::{
-            CoreCommonError, CoreError, CoreExecutionError, CoreInstantiationError, CoreLoadError,
-            InstanceError, VmError, WasmEdgeError,
+            CoreCommonError, CoreError, CoreExecutionError, CoreLoadError, InstanceError, VmError,
+            WasmEdgeError,
         },
-        Config, FuncType, Function, ImportInstance, ImportObject, Loader, Module, Store,
-        WasiModule, WasmValue,
+        Config, Loader, Module, Store, WasmValue,
     };
     #[cfg(target_os = "linux")]
     use crate::{utils, ImportModule, WasmEdgeProcessModule};

--- a/bindings/rust/wasmedge-types/src/error.rs
+++ b/bindings/rust/wasmedge-types/src/error.rs
@@ -62,6 +62,9 @@ pub enum WasmEdgeError {
     NotFoundNulByte(#[from] std::ffi::FromBytesWithNulError),
     #[error("Fail to interpret a sequence of u8 as a string")]
     Utf8(#[from] std::str::Utf8Error),
+
+    #[error("Fail to convert path on Windows: {0}")]
+    WindowsPathConversion(String),
 }
 
 /// The error types for WasmEdge Function.

--- a/bindings/rust/wasmedge-types/src/lib.rs
+++ b/bindings/rust/wasmedge-types/src/lib.rs
@@ -228,6 +228,31 @@ impl From<CompilerOptimizationLevel> for u32 {
         }
     }
 }
+impl From<i32> for CompilerOptimizationLevel {
+    fn from(val: i32) -> CompilerOptimizationLevel {
+        match val {
+            0 => CompilerOptimizationLevel::O0,
+            1 => CompilerOptimizationLevel::O1,
+            2 => CompilerOptimizationLevel::O2,
+            3 => CompilerOptimizationLevel::O3,
+            4 => CompilerOptimizationLevel::Os,
+            5 => CompilerOptimizationLevel::Oz,
+            _ => panic!("Unknown CompilerOptimizationLevel value: {}", val),
+        }
+    }
+}
+impl From<CompilerOptimizationLevel> for i32 {
+    fn from(val: CompilerOptimizationLevel) -> i32 {
+        match val {
+            CompilerOptimizationLevel::O0 => 0,
+            CompilerOptimizationLevel::O1 => 1,
+            CompilerOptimizationLevel::O2 => 2,
+            CompilerOptimizationLevel::O3 => 3,
+            CompilerOptimizationLevel::Os => 4,
+            CompilerOptimizationLevel::Oz => 5,
+        }
+    }
+}
 
 /// Defines WasmEdge AOT compiler output binary format.
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]

--- a/bindings/rust/wasmedge-types/src/lib.rs
+++ b/bindings/rust/wasmedge-types/src/lib.rs
@@ -32,6 +32,23 @@ impl From<RefType> for u32 {
         }
     }
 }
+impl From<i32> for RefType {
+    fn from(value: i32) -> Self {
+        match value {
+            112 => RefType::FuncRef,
+            111 => RefType::ExternRef,
+            _ => panic!("[wasmedge-types] Invalid WasmEdge_RefType: {:#X}", value),
+        }
+    }
+}
+impl From<RefType> for i32 {
+    fn from(value: RefType) -> Self {
+        match value {
+            RefType::FuncRef => 112,
+            RefType::ExternRef => 111,
+        }
+    }
+}
 
 /// Defines WasmEdge value types.
 #[derive(Debug, Copy, Clone, Eq, PartialEq)]
@@ -139,6 +156,23 @@ impl From<u32> for Mutability {
     }
 }
 impl From<Mutability> for u32 {
+    fn from(value: Mutability) -> Self {
+        match value {
+            Mutability::Const => 0,
+            Mutability::Var => 1,
+        }
+    }
+}
+impl From<i32> for Mutability {
+    fn from(value: i32) -> Self {
+        match value {
+            0 => Mutability::Const,
+            1 => Mutability::Var,
+            _ => panic!("[wasmedge-types] Invalid WasmEdge_Mutability: {:#X}", value),
+        }
+    }
+}
+impl From<Mutability> for i32 {
     fn from(value: Mutability) -> Self {
         match value {
             Mutability::Const => 0,
@@ -260,6 +294,20 @@ pub enum ExternalInstanceType {
 }
 impl From<u32> for ExternalInstanceType {
     fn from(value: u32) -> Self {
+        match value {
+            0 => ExternalInstanceType::Func(FuncType::default()),
+            1 => ExternalInstanceType::Table(TableType::default()),
+            2 => ExternalInstanceType::Memory(MemoryType::default()),
+            3 => ExternalInstanceType::Global(GlobalType::default()),
+            _ => panic!(
+                "[wasmedge-types] Invalid WasmEdge_ExternalType: {:#X}",
+                value
+            ),
+        }
+    }
+}
+impl From<i32> for ExternalInstanceType {
+    fn from(value: i32) -> Self {
         match value {
             0 => ExternalInstanceType::Func(FuncType::default()),
             1 => ExternalInstanceType::Table(TableType::default()),

--- a/bindings/rust/wasmedge-types/src/lib.rs
+++ b/bindings/rust/wasmedge-types/src/lib.rs
@@ -89,6 +89,35 @@ impl From<ValType> for u32 {
         }
     }
 }
+impl From<i32> for ValType {
+    fn from(value: i32) -> Self {
+        match value {
+            127 => ValType::I32,
+            126 => ValType::I64,
+            125 => ValType::F32,
+            124 => ValType::F64,
+            123 => ValType::V128,
+            112 => ValType::FuncRef,
+            111 => ValType::ExternRef,
+            64 => ValType::None,
+            _ => panic!("[wasmedge-types] Invalid WasmEdge_ValType: {:#X}", value),
+        }
+    }
+}
+impl From<ValType> for i32 {
+    fn from(value: ValType) -> Self {
+        match value {
+            ValType::I32 => 127,
+            ValType::I64 => 126,
+            ValType::F32 => 125,
+            ValType::F64 => 124,
+            ValType::V128 => 123,
+            ValType::FuncRef => 112,
+            ValType::ExternRef => 111,
+            ValType::None => 64,
+        }
+    }
+}
 
 /// Defines the mutability property of WasmEdge Global variables.
 ///

--- a/bindings/rust/wasmedge-types/src/lib.rs
+++ b/bindings/rust/wasmedge-types/src/lib.rs
@@ -255,6 +255,23 @@ impl From<CompilerOutputFormat> for u32 {
         }
     }
 }
+impl From<i32> for CompilerOutputFormat {
+    fn from(val: i32) -> CompilerOutputFormat {
+        match val {
+            0 => CompilerOutputFormat::Native,
+            1 => CompilerOutputFormat::Wasm,
+            _ => panic!("Unknown CompilerOutputFormat value: {}", val),
+        }
+    }
+}
+impl From<CompilerOutputFormat> for i32 {
+    fn from(val: CompilerOutputFormat) -> i32 {
+        match val {
+            CompilerOutputFormat::Native => 0,
+            CompilerOutputFormat::Wasm => 1,
+        }
+    }
+}
 
 /// Defines WasmEdge host module registration enum.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]

